com.unity.netcode.gameobjects/Runtime/SceneManagement/ScenesInBuild.cs

```diff
@@ -0,0 +1,252 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// This replaces the need to register scenes and contains the Scenes in Build list (as strings)
+    /// Scenes are ordered identically to the Scenes in Build list indices values
+    /// For refined control over which scenes can be loaded or unloaded during a netcode game session,
+    /// use the <see cref="NetworkSceneManager.VerifySceneBeforeLoading"/> event to add additional
+    /// constraints over which scenes are considered valid.
+    /// In order for clients to get this notification you must subscribe to the <see cref="NetworkSceneManager.OnSceneVerificationFailed"/> event.
+    /// </summary>
+    public class ScenesInBuild : ScriptableObject
+    {
+        [HideInInspector]
+        [SerializeField]
+        internal List<string> Scenes;
+
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+
+        /// <summary>
+        /// Determines if we are running a unit test
+        /// In DEVELOPMENT_BUILD we only check for the InitTestScene
+        /// </summary>
+        private static bool IsRunningUnitTest()
+        {
+#if UNITY_EDITOR
+            if (!EditorApplication.isPlaying)
+            {
+                return false;
+            }
+#endif
+            // For scenes in build, we have to check whether we are running a unit test or not each time
+            return SceneManager.GetActiveScene().name.StartsWith("InitTestScene");
+        }
+
+        /// <summary>
+        /// Assures the ScenesInBuild asset exists and if in the editor that it is always up to date
+        /// </summary>
+        internal static void SynchronizeOrCreate(NetworkManager networkManager)
```

Comment:
we had a conversation with Noel and I'd like to try killing this `NetworkManager` dependency injection from the entire `ScenesInBuild` implementation because it's effectively `NetworkManager` to `ScenesInBuild` and `ScenesInBuild` to `NetworkManager` circular dependency (and circular dependency is an anti-pattern).

I might fail but it might be a fun experiment :P

Reply:
Most recent commit addresses this issue by taking a suggestion made by @jeffreyrainy and making a 3rd class to handle the creation and synchronization.

Reply:
Most recent updates remove the ScenesInBuild asset generation process completely.